### PR TITLE
Financing statement migration

### PR DESCRIPTION
**Pull request by jguertin**
Create tables in the target data model which cover `documents` in the original structure

### CODE REVIEW

**Review by WalterMoar — APPROVED** [review]

**Review by gh2os — APPROVED** [review]

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+def upgrade():
+    registration_number_sequence = sa.schema.Sequence('reg_number_seq', start=10000001)
+    op.execute(sa.schema.CreateSequence(registration_number_sequence))
+
+    op.create_table(
+        'financing_statement',
+        sa.Column('reg_number', sa.String(length=10), primary_key=True, nullable=False),
+        sa.Column('reg_type_cd', sa.CHAR(length=2), sa.ForeignKey('registration_type.reg_type_cd'), nullable=False),
+        sa.Column('status', sa.CHAR(length=1), nullable=False),
+        sa.Column('life', sa.Integer, nullable=False),
```
Comment (bryan-gilbert) [inline ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:29]: Curious .. what is life?

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        sa.Column('expiry_date', sa.Date),
+        sa.Column('surrender_date', sa.Date),
+        sa.Column('lien_value', sa.String(length=15)),
```
Comment (bryan-gilbert) [inline ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:32]: A string and not a number?  is this descriptive or to represent a dollar value.  Will this value be computed from collateral?

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        sa.Column('type_claim', sa.CHAR(length=2)),
+        sa.Column('crown_charge_act', sa.CHAR(length=2), sa.ForeignKey('crown_charge_type.crown_charge_type_cd')),
+        sa.Column('crown_charge_other', sa.String(length=70)),
+        sa.Column('trust', sa.BOOLEAN),
+        sa.Column('discharged', sa.BOOLEAN),
```
Comment (bryan-gilbert) [inline ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:37]: Do the properties such as discharged, renewed and expiry date change over the life of a record in this table?

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        sa.Column('renewed', sa.BOOLEAN),
+        sa.Column('ver_bypassed', sa.BOOLEAN),
+        sa.Column('last_update_timestamp', sa.DateTime, server_default=sa.text('NOW()')),
+        sa.Column('user_id', sa.String(length=8), nullable=False)
```
Comment (bryan-gilbert) [inline ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:41]: who is user?  is this a foreign key into another table?

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        sa.Column('change_type_cd', sa.CHAR(length=2), sa.ForeignKey('registration_change_type.change_type_cd'),
+                  nullable=True),
+        sa.Column('reg_date', sa.DateTime, server_default=sa.text('NOW()'), nullable=False),
+        sa.Column('life', sa.Integer, nullable=False),
+        sa.Column('crown_charge_act', sa.CHAR(length=2), sa.ForeignKey('crown_charge_type.crown_charge_type_cd')),
+        sa.Column('crown_charge_other', sa.String(length=70)),
+        sa.Column('description', postgresql.TEXT),
+        sa.Column('document_number', sa.CHAR(length=8), nullable=False),
+        sa.Column('ims_billing_number', sa.String(length=8)),
+        sa.Column('ims_user_id', sa.String(length=32))
```
Comment (bryan-gilbert) [inline ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:58]: Do we know enough to design the payment system?  would these be better added later?

**Review by bryan-gilbert — COMMENTED** [review]
non-blocking questions

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py (hunk 1 above)
Comment (jguertin) [reply ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:29]: The life of the fs in years

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py (hunk 2 above)
Comment (jguertin) [reply ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:32]: It's a free form field.  You can enter `100 Dollars`

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py (hunk 3 above)
Comment (jguertin) [reply ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:37]: Yes, they may

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py (hunk 4 above)
Comment (jguertin) [reply ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:41]: This is not yet clearly defined.  Not a foreign key as of right now.

File: ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py (hunk 5 above)
Comment (jguertin) [reply ppr-api/migrations/versions/fc3c53132a9b_create_financing_statement_document_tables.py:58]: These fields prefixed with `ims_` are to indicate that they are for data from the old system.  Historical.

**Pull request merged**